app/org/sagebionetworks/bridge/play/controllers/SubpopulationController.java

```diff
@@ -59,13 +67,21 @@ public Result getSubpopulation(String guid) {
         Subpopulation subpop = subpopService.getSubpopulation(session.getStudyIdentifier(), subpopGuid);
         return okResult(subpop);
     }
-    public Result deleteSubpopulation(String guid) {
-        UserSession session = getAuthenticatedSession(DEVELOPER);
+    public Result deleteSubpopulation(String guid, String physicalDeleteString) {
+        UserSession session = getAuthenticatedSession();
+        if (!session.getUser().isInRole(DELETE_ROLES)) {
+            throw new UnauthorizedException();
+        }
+        
         SubpopulationGuid subpopGuid = SubpopulationGuid.create(guid);
         
-        subpopService.deleteSubpopulation(session.getStudyIdentifier(), subpopGuid);
+        // Only admins can request a physical delete.
+        boolean physicalDelete = (session.getUser().isInRole(ADMIN) && "true".equals(physicalDeleteString));
```

Comment:
Is it really okay to silently degrade to logical delete for Developer accounts? I can see this leading to a bug somewhere down the line. I'd recommend throwing UnauthorizedException for Developer accounts submitting physical deletes. (This is another point in favor of having separate URLs.)